### PR TITLE
Many botany QoL changes.

### PR DESCRIPTION
**Pull request by FunnyMan3595**
## What Does This PR Do
Many QoL changes, see changelog.
Standardizes the code on "nutrient" for plants, as shown to the user, rather than overloading "nutriment".
(Split from #24342.)

## Why It's Good For The Game
Quality of life is good.

## Images of changes
image
image

## Testing
* Pulled the core stats off of tower caps and placed them onto wheat.
* Scanned an empty tray, saw EZ Nutrient.
* Became a ghost, toggled plant analyzer, clicked on trays, plants, and seeds.
* Grabbed Mutrient from a vendor, used on tray, scanned, saw Mutrient.
* Planted grass in Mutrient and EZ Nutrient. Plant in Mutrient mutated, plant in EZ did not.
* Biogenerator recycled a grass stalk and a grass seed.
* Cmagged a plant data disk, put it back in the plant DNA manipulator.
* Shot skrell with the floral somatoray in both modes, did minor cellular damage.
* Shot a diona with the floral somatoray in mutate mode, gave it Empath.
* Punched a diona in the chest, then shot it there with the floral somatoray in yield mode. Bruise healed.
* Extracted seeds from 10 default wheat at once.
* Dumped a bunch of seeds into the seed extractor. Searched. Sorted.
* Shot a tray with the floral somatoray, at point blank, in both modes.
* Planted 7 plots of starthistle, got 7 batches of starthistle seeds.
* Planted 8 plots of default Ambrosia Gaia. Waited for the weed indicator to show before weeding. All 8 survived to harvest.

## Changelog
:cl:
add: There's an additional ability to extract all core plant stats onto one data disk. When using this mode, core plant stats no longer have caps based on part level, but part level will determine how many seeds with the same stats you need to produce a usable core stats disk (1 to 5).
add: The plant analyzer now reports next harvest time and nutrient in use.
add: Observers now have access to a plant analyzer.
add: Mutrient has been added, duplicating old EZ (green, causes stat mutations)
tweak: EZ-Nutrient (and the tray default) no longer causes plant stat mutations.
tweak: EZ-Nutrient is now brown, and slightly less toxic to people.
tweak: Descriptions for plant nutrient chemicals and their jugs have been adjusted to match the changes.
tweak: The biogenerator will now accept all the same items as a plant bag.
tweak: The biogenerator's time to process is now based around the amount of biomass produced, not the item count. This generally makes the biogenerator faster, especially with small items. Process time is capped at 20s, the same as a full, unupgraded biogenerator before this.
tweak: Cmagged plant data disks now show as "nuclear authentication disk?" in the plant DNA manipulator.
tweak: The floral somatoray now does mild cellular damage to non-dionae.
tweak: The floral somatoray in mutation mode now does a short stun and a genetic mutation (usually bad, never monkey) to dionae.
tweak: The floral somatoray in yield mode now slightly heals dionae.
tweak: Seeds no longer spread quite as far on creation. This should make it somewhat easier to continue clicking the seed extractor when processing multiple plants.
tweak: The seed extractor UI has been overhauled.
fix: The floral somatoray can now be used on plant trays at point blank.
fix: The floral somatoray now produces an on-hit effect in both modes, not just one.
fix: Starthistle no longer randomly refuses to drop anything on harvest.
remove: Weed invasion can no longer replace plants without warning.
/:cl:

### CODE REVIEW

**Comment by FunnyMan3595** [comment]
> So this will still clamp tobacco yield (10) to 5?

The intent of #24122 was to clamp ALL yield to 5. The PR just forgot to update starting seeds to respect the new limit.

It's technically a bug, but since #24342 plans to return the cap to 10, I'm not too fussed about fixing it.

File: code/modules/hydroponics/growninedible.dm [hunk 1]
```diff
@@ -69,3 +64,17 @@
 	if(seed.get_gene(/datum/plant_gene/trait/glow/shadow))
 		return
 	set_light(0)
+
+/obj/item/grown/proc/send_plant_details(mob/user)
+	var/msg = "<span class='info'>This is \a <span class='name'>[src]</span>\n"
```
Comment (Burzah) [inline code/modules/hydroponics/growninedible.dm:69]:
```suggestion
	var/msg = "<span class='info'>This is \a </span><span class='name'>[src]</span>\n"
```

File: code/modules/hydroponics/hydroponics_tray.dm [hunk 2]
```diff
@@ -1021,3 +1041,28 @@
 
 	if(reagent_source) // If the source wasn't composted and destroyed
 		reagent_source.update_icon()
+
+
+/obj/machinery/hydroponics/proc/send_plant_details(mob/user)
+	if(myseed)
+		to_chat(user, "*** <B>[myseed.plantname]</B> ***")
```
Comment (Burzah) [inline code/modules/hydroponics/hydroponics_tray.dm:1048]:
```suggestion
		to_chat(user, "*** <b>[myseed.plantname]</b> ***")
```

File: code/modules/hydroponics/hydroponics_tray.dm [hunk 3]
```diff
@@ -1021,3 +1041,28 @@
...
+		to_chat(user, "- Plant Age: <span class='notice'>[age]</span>")
+		var/next_harvest = (age <= myseed.maturation ? myseed.maturation : lastproduce) + myseed.production
+		to_chat(user, "- Next Harvest At: <span class='notice'>[next_harvest]</span>")
+		var/list/text_string = myseed.get_analyzer_text()
+		if(text_string)
+			to_chat(user, text_string)
+	else
+		to_chat(user, "<B>No plant found.</B>")
```
Comment (Burzah) [inline code/modules/hydroponics/hydroponics_tray.dm:1056]:
```suggestion
		to_chat(user, "<b>No plant found.</b>")
```

**Review by Burzah — CHANGES_REQUESTED** [review]

**Review by S34NW — COMMENTED** [review]
I've had the highest assurances this works fine, if it does not Funnyman gets lashes

**Review by S34NW — APPROVED** [review]

**Review by Burzah — APPROVED** [review]

**Pull request merged**